File: Data Structures & Algorithms/is-anagram/submission-0.swift
class Solution {
    func isAnagram(_ s: String, _ t: String) -> Bool {
        if s.count != t.count {
            return false
        }
        var freq1 = [Character: Int]()
        var freq2 = [Character: Int]()
        
        for (c1, c2) in zip(s, t) {
            freq1[c1, default: 0] += 1
            freq2[c2, default: 0] += 1
        }

        for c in s {
            if freq1[c] != freq2[c] {
                return false
            }
        }
        

        return true
    }
}

File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans(n+1);
        for (int i = 0; i <= n; i++) {
            int count = 0;
            int curr = i;
            while(curr) {
                count += curr%2;
                curr /= 2;
            }
            ans[i] = count;
        }

        return ans;
    }
};
